### PR TITLE
Fix standardstatecorrection

### PR DESCRIPTION
**Pull request by fjclark**
# Is this pull request to fix a bug, or to introduce new functionality?

The current version of `standardstatecorrection` is broken because I failed to include all required changes in the original PR (due to a mix up with personal branches) - this PR adds these changes. I have tested the updated script with an old trajectory and obtained exactly the same correction as previously (using the version [here](https://github.com/fjclark/Sire/blob/mdr_bugfixed_from_boresch/wrapper/Tools/StandardState.py)).

## If this is to fix a bug...

This pull request fixes issue https://github.com/OpenBioSim/sire/issues/80 . 

* I confirm that I have merged the latest version of `devel` into this branch before issuing this pull request (e.g. by running `git pull origin devel`): [y]
* I confirm that I have added a changelog entry to the changelog (we will add a link to this PR as part of the review): [y]
* I confirm that I have permission to release this code under the GPL3 license: [y]

## Suggested reviewers:
@chryswoods

### CODE REVIEW

**Comment by chryswoods** [comment]
The failure is due to an issue with the conda recipe. I’ve fixed it in the other PR currently building. I’ll apply the fix to your branch when I accept and merge the other PR (assuming it all builds ok)

**Comment by fjclark** [comment]
Great, thanks for letting me know.

**Comment by chryswoods** [comment]
I've fixed things in devel and have updated your branch / PR. It is now building. Hopefully it will all be green ticks tomorrow morning :-)

**Review by chryswoods — DISMISSED** [review]
I'm happy for this to be merged once @jmichel80 is happy.

**Comment by fjclark** [comment]
Thanks @chryswoods. I've also just added a test. This uses very few integration points and a short trajectory to get it to run in under 3 seconds - as a result the correction is huge and incorrect, but at least it ensures that the code runs.

File: wrapper/Tools/StandardState.py [hunk 1]
```diff
@@ -177,265 +205,306 @@ def genOrientations(restr_dict, norientations=5):
...
+    kbT = kb * T
+    beta = 1 / kbT
 
     Ztot = 0.0
     Uavg = 0
 
+    # If norient <= 3, then there will be no orientations with any weight.
+    if norient.val <= 3:
+        print("Error, norient must be > 3. Abort.")
+        sys.ext(-1)
```
Comment (jmichel80) [inline wrapper/Tools/StandardState.py:270]: typo / sys.exit(-1)

File: wrapper/Tools/StandardState.py [hunk 2]
```diff
@@ -177,265 +205,306 @@ def genOrientations(restr_dict, norientations=5):
...
             restr_dict[pairs][1].append(coord1)
         current_frame += step_frame
 
-    #now restr_dict has:
-    #restr_dict[lig,host]=[ [req,K,D], [ [coords]...] ,[ [coords],...] ]
+    # now restr_dict has:
+    # restr_dict[lig,host]=[ [req,K,D], [ [coords]...] ,[ [coords],...] ]
     print("Calculating average coordinates for restrained atoms")
-    #import pdb; pdb.set_trace()
+    # import pdb; pdb.set_trace()
```
Comment (jmichel80) [inline wrapper/Tools/StandardState.py:393]: delete from script

**Comment by jmichel80** [comment]
The CI failed for python 3.11 on windows with the following message

```
..\_test_env\Lib\subprocess.py:1538: FileNotFoundError
=========================== short test summary info ===========================
ERROR tests/somd/test_standardstatecorrection.py::test_standard_state_correction - FileNotFoundError: [WinError 2] The system cannot find the file specified
```
builds on other OSs completed without error.

**Comment by jmichel80** [comment]
Quick [googling](https://stackoverflow.com/questions/33560364/python-windows-parsing-command-lines-with-shlex) suggests that shlex may not be reliable on windows.  
@chryswoods would you have suggestions for a portable command line parser that could be used to rewrite this [test](https://github.com/OpenBioSim/sire/blob/devel/tests/somd/test_standardstatecorrection.py) ?

**Comment by chryswoods** [comment]
I've pushed a fix for this test to `devel`. There were a few things to fix:

1. The test needed marking as being slow, so that it is skipped for quick tests.
2. shlex.split is unreliable and there is no good option on Windows the works all the time in CI. As such, we disable tests that use shlex if this is run on Windows, e.g. see tests/somd/test_somd.py.
3. The test needs to be marked to be skipped if its dependencies aren't available. These are scipy and wget. We can remove the wget dependency if the test is migrated to use the same remote download code that we use to download bzipped structures. Maybe @mb2055 could take a look and update the test to download the DCD file from https://sire.openbiosim.org/m as it does for other structure files?

**Comment by chryswoods** [comment]
I've back-ported this fix as part of the #82 PR (back-port all to `main`). I've also run a new build/deploy GH Action on devel just for Windows, so that the Windows devel package will be uploaded by tomorrow morning.

**Pull request merged**